blog/2024-02-12-async_update_entry.md

```diff
@@ -0,0 +1,20 @@
+---
+author: J. Nick Koston
+authorURL: https://github.com/bdraco
+title: Altering config entries
+---
+
+In Home Assistant 2024.3, modifications to a `ConfigEntry` should use `hass.config_entries.async_update_entry`. Directly setting attributes on the `ConfigEntry` object is deprecated and will fail in version 2024.10 and later. There is no deprecation period for directly setting `unique_id` or `entry_id` on the `ConfigEntry` as doing so will corrupt the internal state.
+
+The following attributes must now be set via `hass.config_entries.async_update_entry`:
+
+- data
+- minor_version
+- options
+- pref_disable_new_entities
+- pref_disable_polling
+- title
+- unique_id
```

Comment:
This doesn't list the `entry_id`, but would it better to avoid the listing altogether and state that all modifications to config entries should be done through `async_update_entry`?

Reply:
The `entry_id` should never be changed externally as its fully generated internally and changing it will corrupt internal state. I don't think anybody does that in production.. at least I hope.

Reply:
Yeah, I just raised it as it was explicitly mentioned in the preamble but not here. Otherwise, it feels sane to me without having read through the discussions in the parent PR.

Reply:
Thats a good point. I took `entry_id` out of the top section as its confusing and nobody should be doing that anyways.